core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,128 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(self.user_id), exp_ids)
 
-        # The list still remains the same.
+        # Also if we try adding another exploration at no specific location,
+        # it shouldn't be added as the list length would exceed
+        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
+        learner_playlist_services.mark_exploration_to_be_played_later(
+            self.user_id, 'SAMPLE_EXP_ID')
         self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), exp_ids)
+            self._get_all_learner_playlist_exp_ids(self.user_id), exp_ids)
+
+    def test_adding_subscribed_collections_to_playlist_does_not_add_again(self):
+        # Subscribe to collection.
+        subscription_services.subscribe_to_collection(
+            self.user_id, 'SUBSCRIBED_ID')
 
-    def test_mark_collection_to_be_played_later(self):
+        # Now if we try to add the same collection, it shouldn't be added.
+        learner_progress_services.add_collection_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(self.user_id), [])
 
-        # Add a collection to the learner playlist.
+        learner_playlist_services.remove_collection_from_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Test adding the collection_id if it is already in
+        # learner_playlist.collection_ids
+        # Add two collections to the learner playlist.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_0)
-        self.assertEqual(
-            self._get_all_learner_playlist_collection_ids(
-                self.user_id), [self.COL_ID_0])
-
-        # Add another collection.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_1)
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(
                 self.user_id), [self.COL_ID_0, self.COL_ID_1])
 
-        # Add the first collection in a different position.
+        # Add the first collection to the second position.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_0,
             position_to_be_inserted=1)
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(
                 self.user_id), [self.COL_ID_1, self.COL_ID_0])
 
-        # Add the first collection in the first position.
-        learner_progress_services.add_collection_to_learner_playlist(
-            self.user_id, self.COL_ID_0,
-            position_to_be_inserted=0)
-        self.assertEqual(
-            self._get_all_learner_playlist_collection_ids(
-                self.user_id), [self.COL_ID_0, self.COL_ID_1])
-
-        # Add a collection to the in progress list of the user.
```

Comment:
Why is this removed?

Reply:
This tests the same action as "#Add the first collection to the second position." Both involve shifting the learner playlist. Should I add it back?

Reply:
See comment above.

Reply:
Got it.